dal/UserDal.js

```diff
@@ -0,0 +1,83 @@
+const Dal = require('./Dal');
+const User = require('../models/User');
+const SessionService = require('../services/Session');
+
+class UserDal extends Dal {
+    constructor() {
+        super();
+    }
+
+    async createUser(user) {
+        try {
+            await this.mongo.connect();
```

Comment:
why do you need to connect to mongo for each operation, shouldn't you have some connection manager?